### PR TITLE
Remove "Edit on Github" button

### LINKED ISSUE #3494: `Edit on GitHub` links in many reference pages do not work
## Issue description

This issue was originally reported by @fukatani.

When I clicked the `Edit on GitHub` in some [reference page](https://optuna.readthedocs.io/en/latest/reference/visualization/generated/optuna.visualization.plot_pareto_front.html), I got 404 errors.

image

This is because rst files are dynamically generated in CI jobs, and they are not in the github repository. So, I think the links seems to be noise for readers.

## Expected behavior

Remove `Edit on GitHub` links from the reference pages.

## Changes

The source link can be disabled by the [html_show_sourcelink](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_show_sourcelink) option in `config.py`. In my environment, it removed the `Edit on GitHub` links but preserved the source links near the method names.

image

## Alternative approach

The [`display_github` option in `html_context`](https://docs.readthedocs.io/en/stable/guides/edit-source-links-sphinx.html) is also related to this issue, but `View page source` links are shown instead of `Edit on GitHub` links. The destination of the link is rst file that contains only a list of methods or functions.  It may confuse the developers who want to edit the content of the reference, so I think we can simply remove the links.

image

### PR DESCRIPTION
**Pull request by cfkazu**
## Motivation

Solve https://github.com/optuna/optuna/issues/3494
To remove  `Edit on GitHub` and `[source]`

## Description of the changes

Simply add "html_show_sourcelink=False" and remove extention "sphinx.ext.viewcode"

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -51,7 +51,6 @@
     "sphinx.ext.intersphinx",
     "sphinx.ext.mathjax",
     "sphinx.ext.napoleon",
-    "sphinx.ext.viewcode",
```
[the remark this answers is not in the document]
Comment (cfkazu) [reply docs/source/conf.py:54]: Thank you for your comments.
I noticed my misreading about the point of “[source]”. I’ll fix it.

File: docs/source/conf.py [hunk 2]
```diff
@@ -202,3 +201,6 @@
 plotly_formats = ["html"]
 plotly_html_show_formats = False
 plotly_html_show_source_link = False
+
+# remove sourcelink
+html_show_sourcelink = False
```
[the remark this answers is not in the document]
Comment (cfkazu) [reply docs/source/conf.py:206]: I agree with your opinion.

File: docs/source/conf.py [hunk 3]
```diff
@@ -103,6 +103,9 @@
 html_static_path = ["_static"]
 html_css_files = ["css/custom.css"]
 
+# Remove "Edit on GitHub"
```
Comment (toshihikoyanase) [inline docs/source/conf.py:106]: Could you add a period to follow the coding-style conventions?
https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#block-and-inline-comments-are-supposed-to-be-sentences

**Review by toshihikoyanase — COMMENTED** [review]
`Edit on GitHub` links are removed as expected. Thank you.

I have a small comment.
image

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Comment by nzw0301** [comment]
We can see `Edit on Github button` again on the latest stable docs: https://optuna.readthedocs.io/en/stable/tutorial/index.html.

**Pull request merged**